api_pagopa.yaml

```diff
@@ -37,11 +37,50 @@ paths:
           description: There was an error in retrieving the user data.
           schema:
             $ref: "#/definitions/ProblemJson"
+  "/user-extended":
```

Comment:
@francescopersico do you agree with the naming here ?

Reply:
@Undermaken what about the naming here ?

Reply:
@gunzip @francescopersico 
since `/user-extended` is a kind of evolution of **/user** and it will replace it (isn't it?), could it have more sense to create a new version of user `/v2/user` or `&user=v2`?

Otherwise, in future, we will need to keep `/user-extended` as official api to get an user

Reply:
good point but 

- `/v2/user` it's problematic since we would have to change the base api path for all endpoints to `/api/v2`
- with `&user=v2` do you mean `/user&version=2` ?

so I see two options

1. we change the base path to `/v2` for all the endpoints and force all the clients to upgrade it (in the current scenario we have just one client which is the app)
2. we use /user?version=2 which let us mark the query parameter as obsolete at some point in the future and does not requires client to update their base path

since this should be into the guidelines, I'd cc @cloudify

Reply:
@gunzip on [2] scenario how we can map the API spec semantically with the expected response?

Reply:
it should return a union of the two types (oneOf). Since it's not supported in swagger 2.x (openapi supports it)
we have hacked it back into io-utils: https://github.com/teamdigitale/io-utils/blob/9bdd9307177cc25dd5cc70304c5e2c8bf17a1332/src/__tests__/api.yaml#L60

Reply:
Ok, but this spec is not strict where the missing version returns a `PagoPAUser` and version `20200114` returns `ExtendedPagoPAUser` instead.

Reply:
yes, in any case the caller must discriminate between the returned type. another option is to make a backward compatible change. question: if we replace the current endpoint with this one, why this should break compatibility ?

Reply:
No one compatibility issues. I'll proceed with the suggested implementation.

Reply:
Fixed in 193b0d1